File: line-sdk/src/main/java/com/linecorp/linesdk/openchat/OpenChatParameters.java
package com.linecorp.linesdk.openchat;

import org.json.JSONException;
import org.json.JSONObject;

public class OpenChatParameters {
    private static int DEFAULT_CATEGORY_ID = 17; // Game
    private OpenChatParameters() {}

    String name;
    String description;
    String creatorDisplayName;
    int categoryId = DEFAULT_CATEGORY_ID;
    Boolean isSearchable = true;

    public String toJsonString() {
        JSONObject jsonObject = new JSONObject();

        try {
            jsonObject.put("name", name);
            jsonObject.put("description", description);
            jsonObject.put("creatorDisplayName", creatorDisplayName);
            jsonObject.put("category", categoryId);
            jsonObject.put("allowSearch", isSearchable);
        } catch (JSONException exception) {
            return "{}";
        }

        return jsonObject.toString();
    }

    public static class Builder  {
        private OpenChatParameters openChatParameters = new OpenChatParameters();

        public Builder setName(String name) {
            if (name == null || name.length() > 50) {
                throw new IllegalArgumentException("String size needs to be less or equal to 50");
            }
            openChatParameters.name = name;

            return this;
        }

        public Builder setDescription(String description) {
            if (description == null || description.length() > 200) {
                throw new IllegalArgumentException("String size needs to be less or equal to 200");
            }
            openChatParameters.description = description;

            return this;
        }

        public Builder setCreatorDisplayName(String creatorDisplayName) {
            if (creatorDisplayName == null || creatorDisplayName.length() > 50) {
                throw new IllegalArgumentException("String size needs to be less or equal to 50");
            }
            openChatParameters.creatorDisplayName = creatorDisplayName;

            return this;
        }

        public Builder setCategoryId(int categoryId) {
            openChatParameters.categoryId = categoryId;
            return this;
        }

        public Builder setIsSearchable(Boolean isSearchable) {
            openChatParameters.isSearchable = isSearchable;
            return this;
        }

        public OpenChatParameters build() {
            if (openChatParameters.name == null
                        || openChatParameters.name.isEmpty()
                        || openChatParameters.creatorDisplayName == null
                        || openChatParameters.creatorDisplayName.isEmpty()) {
                throw new IllegalArgumentException("Parameters are not correctly set up");
            }

            return openChatParameters;
        }
    }
}

File: line-sdk/src/main/java/com/linecorp/linesdk/openchat/OpenChatParameters.kt
package com.linecorp.linesdk.openchat

import org.json.JSONException
import org.json.JSONObject

class OpenChatParameters(
    private val name: String,
    private val description: String,
    private val creatorDisplayName: String,
    private val category: OpenChatCategory = OpenChatCategory.Game,
    private val isSearchable: Boolean = true
) {
    init {
        require(name.isNotEmpty() && name.length <= 50) { "String size needs to be less or equal to 50" }
        require(description.length <= 200) { "String size needs to be less or equal to 200" }
        require(creatorDisplayName.isNotEmpty() && creatorDisplayName.length <= 50) { "String size needs to be less or equal to 50" }
    }

    fun toJsonString(): String = try {
        JSONObject().apply {
            put("name", name)
            put("description", description)
            put("creatorDisplayName", creatorDisplayName)
            put("category", category.id)
            put("allowSearch", isSearchable)
        }.toString()
    } catch (exception: JSONException) {
        "{}"
    }
}
